beacon-chain/blockchain/head.go

```diff
@@ -122,7 +122,7 @@ func (s *Service) saveHead(ctx context.Context, newHeadRoot [32]byte, headBlock
 		reorgDistance.Observe(float64(dis))
 		reorgDepth.Observe(float64(dep))
 
-		isOptimistic, err := s.IsOptimistic(ctx)
+		isOptimistic, err := s.ForkChoicer().IsOptimistic(newHeadRoot)
```

Comment:
This function `saveHead`  calls later the function `notifyNewHeadEvent`  which acquires a lock for `IsOptimistic`. It would be nice to check the safety of this